_posts/2022-03-15-boa-release-14.md

```diff
@@ -0,0 +1,87 @@
+
+## Writing a VM
+
+The biggest change in this release has definitely been the switch of the main implementation to a Virtual Machine. The
+JavaScript code first gets parsed into AST, then compiled into op-codes, and finally executed in the VM. This adds some
+extra steps, which, for now, have reduced the performance of the engine. We hope to fix some of these degradations over the course of the next few releases so hang tight. The new VM opens the door for many great features in the future (we're looking at you, async/await), and allows some nicer optimizations in the parsing and compilation phases in which we have already started working (such as [#1829](https://github.com/boa-dev/boa/pull/1829) and
+[#1849](https://github.com/boa-dev/boa/pull/1849)).
+
+This move towards a VM has also brought one great new feature, generator execution ([#1790](https://github.com/boa-dev/boa/pull/1790)).
+
+## String interning
+
+@Razican has done some great work on the string interner for Boa. Interning allows many types to not contain heap-allocated strings, and just contain a `NonZeroUsize` instead, which can then be placed onto the stack.
+
+TODO: @Razican can we add more here?
+
+[Benchmarks](https://github.com/boa-dev/boa/pull/1765#issuecomment-1019474176) improved overall.
+
+## Boa opens financial contributions
+
+In the past months, due to the big and steady progress on Boa, we have received multiple requests to contribute to Boa
+financially. From now on, anyone can contribute with either a recurring donation, or a one-time contribution in our
+[OpenCollective page](https://opencollective.com/boa).
+
+### What does this mean for Boa?
+
+Being able to receive financial contributions won't directly show a change in Boa development, but depending on how much
+funding we get, we are already planning next steps. Having more motivated contributors would be the first step. We all
+want to spend more of our free time in boa, and receiving a financial compensation for it is definitely making the team
+members eager to continue contributing.
+
+But that is not our only idea. We would like to create a reproducible benchmarking platform. Currently, we just use GitHub
+actions for benchmarking, which is useful to get some general ideas on how some changes affect the performance of the
+engine. However, they have a high variability (up to 10%), meaning that we can definitely see changes if they are big, but it's
+impossible to see differences if changes are minimal (1-2%, for example). Having the infrastructure to do proper performance
+testing would require using dedicated CPUs, for example, which could be financed using the Boa Open Collective.
+
+On the development side, there are some very big projects that require deep involvement from the team. Having
+[async/await support](https://github.com/boa-dev/boa/projects/3), or implementing the
+[module system](https://github.com/boa-dev/boa/issues/64) is one of our priorities, and blocks much of our conformance.
```

Comment:
I don't think a module system is a priority for us right now as a runtime using Boa would most likely add this.